WooCommerce/Classes/ViewRelated/Products/Edit Product/Subscription/Trial/SubscriptionTrialViewModel.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import Yosemite
+import Combine
+
+/// ViewModel for `SubscriptionTrialView`
+///
+final class SubscriptionTrialViewModel: ObservableObject {
+    @Published var trialLength: String
+
+    @Published var trialPeriod: SubscriptionPeriod
+
+    @Published var isInputValid: Bool = true
+
+    var trialPeriodDescription: String {
+        switch trialLength {
+        case "1":
+            return trialPeriod.descriptionSingular
+        default:
+            return trialPeriod.descriptionPlural
+        }
+    }
+
+    typealias Completion = (_ trialLength: String,
+                            _ trialPeriod: SubscriptionPeriod,
+                            _ hasUnsavedChanges: Bool) -> Void
+    private let onCompletion: Completion
+
+    private let subscription: ProductSubscription
+
+    /// View title
+    ///
+    let title = Localization.title
+
+    init(subscription: ProductSubscription,
+         completion: @escaping Completion) {
+        self.subscription = subscription
+        self.onCompletion = completion
+        self.trialLength = subscription.trialLength
+        self.trialPeriod = subscription.trialPeriod
+
+        Publishers.CombineLatest($trialLength,
+                                 $trialPeriod)
+        .map { trialLength, trialPeriod in
+            guard let length = Int(trialLength) else {
+                return false
+            }
+
+            switch trialPeriod {
+            case .day:
+                return length <= 90
+            case .week:
+                return length <= 52
+            case .month:
+                return length <= 24
+            case .year:
+                return length <= 5
+            }
+        }
+        .assign(to: &$isInputValid)
+    }
+
+    func didTapDone() {
+        onCompletion(trialLength,
+                     trialPeriod,
+                     hasUnsavedChanges())
+    }
+}
+
+private extension SubscriptionTrialViewModel {
+    func hasUnsavedChanges() -> Bool {
+        trialLength != subscription.trialLength || trialPeriod != subscription.trialPeriod
```

Comment:
np and edge case and just want to mention: 

if `trialLength` is zero and user just changes period, then it is also considered as unsaved changes although in practice it's still zero amount of time.

Reply:
Nice catch. Covered this in https://github.com/woocommerce/woocommerce-ios/pull/11173/commits/ea2d2168fe8e551fec9217e10d8a423e63bd58af.